core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/autotelemetry21/custom_telemetry.go

```diff
@@ -66,8 +68,15 @@ func (e *AutomationCustomTelemetryService) Start(ctx context.Context) error {
 					if err != nil {
 						e.lggr.Errorf("Error occurred while getting newestConfigDetails in configDigest loop %s", err)
 					}
+					configChanged := false
+					e.mu.Lock()
 					if newConfigDigest != e.configDigest {
 						e.configDigest = newConfigDigest
+						configChanged = true
+					}
+					e.mu.Unlock()
+
+					if configChanged {
 						e.sendNodeVersionMsg()
 					}
```

Comment:
why change this behaviour of only sending this message on config change and not on regular intervals?

Reply:
In the previous version of the code, we only sent the message when the config had changed: https://github.com/smartcontractkit/chainlink/blob/1a2b7b61cbd22256e4e29e891a74228fa453fc9d/core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/autotelemetry21/custom_telemetry.go#L71

We need to lock the mutex before checking if the config has changed (the if statement); but because the `sendNodeVersionMsg` itself locks the Read lock on the mutex (because it gets called from other places, too), we can't execute the `sendNodeVersionMsg` function while the lock is held. And so, we're having to introduce this additional boolean that gets set to true when the config changes (previously, this is when we would have called `sendNodeVersionMsg`), and then reference that boolean outside the lock to determine if we should call `sendNodeVersionMsg`.

Reply:
ah sorry had missed that `e.sendNodeVersionMsg()` was within the if statement before. Thanks for clarifying